src/services/symbolDisplay.ts

```diff
@@ -496,7 +496,7 @@ namespace ts.SymbolDisplay {
             addNewLineIfDisplayPartsExist();
             if (symbolKind) {
                 pushTypePart(symbolKind);
-                if (!some(symbol.declarations, d => isArrowFunction(d) || (isFunctionExpression(d) || isClassExpression(d)) && !d.name)) {
+                if (symbol && !some(symbol.declarations, d => isArrowFunction(d) || (isFunctionExpression(d) || isClassExpression(d)) && !d.name)) {

```

Comment:
So if `symbol === undefined` we will just leave the name blank? Is that desirable behavior? Should we perhaps instead fail to build the display completely?

Reply:
This code path builds either anonymous or non-anonymous declarations (e.g. `function (x)` or `function y(x)`), so it's already regularly producing non-named symbol displays. In the case where the originating type doesn't have a symbol, it's sort of de facto anonymous anyway so it seems like this won't ever produce any particularly weird output.